### DISCUSSION TITLE
use Trivy to scan a Java compiled program (its imported jar files)

### DISCUSSION THREAD

**Discussion by aturbati** [aquasecurity/trivy#2844] [Q&A] [answered]
I'm new to Trivy (I'm using version 0.31.2 ) and I would like to use it to scan a compiled Java Program. In particular I would like to use Trivy to scan all the Jar files contained in such program (which was compiled using Maven). The result of the compiling of the Java program is a zip file. So I thought I could do (after I've unziped such file) something like:

trivy filesystem --vuln-type library --severity CRITICAL PATH_OF_UNZIPPED_FILE

since in the PATH_OF_UNZIPPED_FILE there are several folders, some of them may contain other folders and subfolders with some jars in them (and that are the files I'm interested that should be scanned).  These jars, inside them, have a pom.xml file.
I've tried as described but Trivy returns almost immediatelly providing no results (I know that some of the jar contains some vulnerability in them) so I'm pretty sure I'm doing something wrong or I do understand how Trivy should be used. 
Anyone has any suggestion on how should I do it?

**Comment by Kausik-A** [comment]
Hey, I was looking to do something similar. Did you figure out if trivy supports security scanning for jar files ?

**Reply by knqyf263** [reply, marked as answer]
Please reference the table. 
https://aquasecurity.github.io/trivy/v0.41/docs/scanner/vulnerability/language/

JAR is not supported in fs scanning, but rootfs. We might be improving the UX, but you need to use rootfs now.
